lib/shared_print/finder.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require "cluster"
+require "services"
+
+Services.mongo!
+
+module SharedPrint
+  # Pass criteria to new(), and
+  # * call clusters()    to yield the matching clusters,
+  # * call commitments() to yield the matching commitments.
+  class Finder
+    attr_reader :organization, :ocn, :local_id, :deprecated, :query
+    def initialize(organization: [], ocn: [], local_id: [], deprecated: false)
+      @organization = organization
+      @ocn = ocn
+      @local_id = local_id
+      @deprecated = deprecated
+
+      # Put together a query based on the criteria gathered.
+      @query = build_query
+    end
+
+    # Yield matching clusters.
+    def clusters
+      Cluster.where(@query).no_timeout.each do |cluster|
+        yield cluster
+      end
+    end
+
+    # Yield matching commitments in matching clusters.
+    def commitments
+      clusters do |cluster|
```

Comment:
`return enum_for(:commitments) unless block_given?`

Then you can just do:
`finder.commitments.to_a`
obviating the need for `def gather`

Same with `def clusters`.

I think.

Reply:
Yeah, nice.